plugin-server/src/cdp/types.ts

```diff
@@ -0,0 +1,119 @@
+import { VMState } from '@posthog/hogvm'
+
+import { ElementPropertyFilter, EventPropertyFilter, PersonPropertyFilter } from '../types'
+
+export type HogBytecode = any[]
+
+// subset of EntityFilter
+export interface HogFunctionFilterBase {
+    id: string
+    name: string | null
+    order: number
+    properties: (EventPropertyFilter | PersonPropertyFilter | ElementPropertyFilter)[]
+}
+
+export interface HogFunctionFilterEvent extends HogFunctionFilterBase {
+    type: 'events'
+    bytecode: HogBytecode
+}
+
+export interface HogFunctionFilterAction extends HogFunctionFilterBase {
+    type: 'actions'
+    // Loaded at run time from Action model
+    bytecode?: HogBytecode
+}
+
+export type HogFunctionFilter = HogFunctionFilterEvent | HogFunctionFilterAction
+
+export interface HogFunctionFilters {
+    events?: HogFunctionFilterEvent[]
+    actions?: HogFunctionFilterAction[]
+    filter_test_accounts?: boolean
+    // Loaded at run time from Team model
+    filter_test_accounts_bytecode?: boolean
+    bytecode?: HogBytecode
+}
+
+export type HogFunctionInvocationContext = {
```

Comment:
Naming things: Is this a context or just globals? 🤔 
```suggestion
export type HogFunctionInvocationGlobals = {
```